### ISSUE TITLE
sticky: extensions

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#294]
:exclamation: ~~please try to NOT start discussions in here, start a new issue instead. ONLY use this thread to report extensions - _thank you_~~

Use this issue for extension announcements: new, gone-to-sh*t, recommendations for adding or dropping in the wiki list [4.1: Extensions](https://github.com/ghacksuserjs/ghacks-user.js/wiki/4.1-Extensions). Stick to privacy and security related items, and do not mention legacy extensions

### :small_orange_diamond: Added Web Extensions

* ... go look on the Wiki page - not listing things twice

### :small_orange_diamond: Pending Web Extensions

* [Negotiator](https://addons.mozilla.org/firefox/addon/negotiator/) | [GitHub](https://github.com/hultqvist/negotiator)
  * doesn't work (yet) full of bugs and errors - ask earthlng - worth revisiting
* [Google search link fix](https://addons.mozilla.org/firefox/addon/google-search-link-fix/) | [GitHub](https://github.com/palant/searchlinkfix)
   * google/yandex. Personally, I use a user script called "google privacy" which has been working for years - I should find a link for it, see if it's maintained, stick in the wiki for user scripts? -Thorin
* [Searchonymous2](https://addons.mozilla.org/firefox/addon/searchonymous2/) | [GitHub](https://github.com/DaAwesomeP/Searchonymous2)
* [Modify Header Value](https://addons.mozilla.org/firefox/addon/modify-header-value/)
* [Referer Control](https://addons.mozilla.org/firefox/addon/referercontrol/)
* [eCleaner (Forget Button)](https://addons.mozilla.org/firefox/addon/ecleaner-forget-button/) | [GitHub](https://github.com/andy-portmen/ecleaner/)
   * NOTE: bulk clean per storage type: eg indexedDB, dom storage etc
   * NOTE: Is using wrong Storage API for settings??? Requires cookies in order to access local dom.
* [mHeaderControl](https://addons.mozilla.org/firefox/addon/mheadercontrol/) | [GitHub](https://github.com/maheee/mHeaderControl)
* [Shape Shifter](https://addons.mozilla.org/firefox/addon/shape-shifter/) | [GitHub](https://github.com/snapper26/shapeshifter)
* [EditThisCookie](https://addons.mozilla.org/firefox/addon/editthiscookieaddon)
* [History AutoDelete](https://addons.mozilla.org/firefox/addon/history-autodelete)
* FoxyProxy
   * [Basic](https://addons.mozilla.org/firefox/addon/foxyproxy-basic)
   * [Standard](https://addons.mozilla.org/firefox/addon/foxyproxy-standard/)
* [WebAPI Manager](https://addons.mozilla.org/firefox/addon/webapi-manager/) | [GitHub](https://github.com/snyderp/web-api-manager)
* [Forget Me Not](https://addons.mozilla.org/firefox/addon/forget_me_not/) | [GitHub](https://github.com/Lusito/forget-me-not)
* [Luminous](https://addons.mozilla.org/firefox/addon/luminous/) | [GitHub](https://github.com/gbaptista/luminous) - JS Events handling
* [3P Request Blocker](https://addons.mozilla.org/firefox/addon/3prb/)

### :small_orange_diamond: Rejected <sup>If you **strongly** disagree, then by all means, bring it up</sup>

* [Canvas Defender](https://addons.mozilla.org/firefox/addon/no-canvas-fingerprinting)
   * nah - see https://github.com/ghacksuserjs/ghacks-user.js/issues/458
* [VTzilla](https://addons.mozilla.org/en-US/firefox/addon/vtzilla/)
   * shitty privacy policy, you have to opt-out of scanning everything, but could be handy for some people, as long as they are aware of the privacy implications

...

**Comment by Atavic** [comment]
[FacebookTrackingRemoval](https://github.com/mgziminsky/FacebookTrackingRemoval) barely tested here (no FB), but I trust the dev.

**Comment by Thorin-Oakenpants** [comment]
Not interested in Extensions that pander to a single site. The adblocking and cosmetics of hiding elements can already be done (uM, uBo .. heck, even Stylus). The link clicking/tracking can also be done not that I have uber investigated it, but rules can be created I'm sure. So unless I'm missing something.. nah

**Comment by Decopi** [comment]
Please, I would like to submit to be added to the list, Peter Snyder add-on (WebApiManager): [https://addons.mozilla.org/en-US/firefox/addon/webapi-manager/](url)

**Comment by Thorin-Oakenpants** [comment]
@Decopi It is already listed above in the `pending` section

**Comment by Atavic** [comment]
[User Agent Switcher](https://github.com/ray-lothian/UserAgent-Switcher/)

- [AMO](https://addons.mozilla.org/firefox/addon/user-agent-string-switcher/)

- [FAQ](https://add0n.com/useragent-switcher.html)

I know current policy for spoofing, so I'll silently keep an eye on this webext.

**Comment by Decopi** [comment]
Another suggestion, ClearURLs, Kevin Roebert:

[https://addons.mozilla.org/en-US/firefox/addon/clearurls/
](url)

The main difference of this add-on, first is an open source, and second it has 65 rules against average of 13 rules of other similar adds-on.

**Comment by 2glops** [comment]
In fact : https://addons.mozilla.org/en-US/firefox/addon/clearurls/

**Comment by Thorin-Oakenpants** [comment]
- added * [Header Editor](https://addons.mozilla.org/firefox/addon/header-editor/) | [GitHub](https://github.com/FirefoxBar/HeaderEditor)
- moved GM to legacy recommendations - the thing is a mess right now

**Comment by Thorin-Oakenpants** [comment]
ClearURLs has a couple of issues
- you need to set an extension permission exception for cookies in order to be able to log. It doesn't use IDB, but somehow the cookie setting blocks storage of log data
- it has web_accessible_resources [1] which on the face of it seems ok. The log is viewed internally, not sure about the alert part. Can someone look into this?
- bit weird: the counter keeps going up and up per page - eg on reddit it blocks 2 items, then go next page and it climbs to 4 etc. It starts from scratch on a new tab+reddit etc. I'd rather have it PER page, not a running total. That way you can see if anything is blocked or not.

[1]
```js
  "web_accessible_resources": [
      "siteBlockedAlert.html",
      "log.html"
  ]
```

**Comment by KevinRoebert** [comment]
Hi, I'am the owner of ClearURLs. I want to answere your questions:


> you need to set an extension permission exception for cookies in order to be able to log. It doesn't use IDB, but somehow the cookie setting blocks storage of log data

We're saving the log with the following two functions:
```
/**
* Function to log all activities from ClearUrls.
* Only logging when activated.
* The log is only temporary saved in the cache and will
* permanently saved with the saveLogOnClose function.
*
* @param beforeProcessing  the url before the clear process
* @param afterProcessing   the url after the clear process
* @param rule              the rule that triggered the process
*/
function pushToLog(beforeProcessing, afterProcessing, rule)
{
    if(logging)
    {
        log.log.push(
            {
                "before": beforeProcessing,
                "after": afterProcessing,
                "rule": rule,
                "timestamp": Date.now()
            }
        );
    }
}

/**
* This function is triggered by the event windows.onRemoved and tabs.onCreated
* and will save the log permanently to the local storage.
* We only save the log anticyclically based on performance.
*/
function saveLog()
{
    if(logging)
    {
        browser.storage.local.get('resetLog', function(data) {
            if(data.resetLog)
            {
                log = {"log": []}; // Delete the old log
                browser.storage.local.set({"resetLog": false});
            }
            else
            {
                browser.storage.local.set({"log": JSON.stringify(log)});
            }
        });
    }
}
```
We're only save a log when the user activate the logging function. From default the logging is disabled, because it is only a feature for debugging and to make suggestions for new rules.


> it has web_accessible_resources [1] which on the face of it seems ok. The log is viewed internally, not sure about the alert part. Can someone look into this?

The altert part is only a html site, that inform the user, that a whole domain was blocked, because of ad-tracking, like doubleclick{dot}net.


> bit weird: the counter keeps going up and up per page - eg on reddit it blocks 2 items, then go next page and it climbs to 4 etc. It starts from scratch on a new tab+reddit etc. I'd rather have it PER page, not a running total. That way you can see if anything is blocked or not.

Yeah, this is a feature-bug :P. We will change this in the next update. So the counter will counts only the blocked fields on a page, not on a tab like now.

I am available for any questions.

**Comment by Thorin-Oakenpants** [comment]
Thanks @KevinRoebert I am not a developer, so excuse my ignorance.

---
Are you using the right storage API, because a cookie permission should not be needed AFAIK. other web extensions do not have this issue storing some local non-IDB data - I guess it depends where. Yes you would need one if it were IDB - see [1406675](https://bugzilla.mozilla.org/show_bug.cgi?id=1406675) and the work around ( https://github.com/ghacksuserjs/ghacks-user.js/wiki/4.1.1-Setting-Extension-Permission-Exceptions ). It's not critical, because I can easily set a site permission exception on the UUID

FYI: WEs can access their own storage regardless of prefs using the [storage API](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/storage/local) (+permission).

> **Although this API is similar to Window.localStorage it is recommended that you don't use Window.localStorage in extension code**. Firefox will clear data stored by extensions using the localStorage API in various scenarios where users clear their browsing history and data for privacy reasons, while data saved using the storage.local API will be correctly persisted in these scenarios.

---
Web Accessible Resources - see [this issue](https://github.com/ghacksuserjs/ghacks-user.js/issues/227) for more info. The UUID which is unique to each install of a web extension, can be exploited in some cases. Other extensions do not require this for their UI popups, etc. Is it possible to do away with them. AFAICT, you're not injecting or doing anything to web content, just blocking requests - right? So we should be safe (told you I am not an expert!).

---
Good to know about the counter

**Comment by Thorin-Oakenpants** [comment]
> The altert part is only a html site, that inform the user, that a whole domain was blocked, because of ad-tracking, like doubleclick{dot}net.

Yeah, not a fan of this. Why do we need to access an external site to be told a domain was blocked

**Comment by Thorin-Oakenpants** [comment]
@KevinRoebert [Here](https://github.com/ghacksuserjs/ghacks-user.js/issues/227#issuecomment-326823250) is a comment by gorhill (the dev of uBo and uMatrix) that shows you how to bypass using web accessible resources - all over my head, but there ya go :)

**Comment by KevinRoebert** [comment]
> Edit: OK, looks like you're injecting that alert into web pages. This could be a problem. Not your fault - it's Mozilla's fault for using a UUID per install and not masking/hiding it properly

I do not directly injecting the alert into web pages. The progress is the following.

1. ClearURLs detects a blocked domain.
2. ClearURLs rewrite the asked url to  `var siteBlockedAlert = browser.extension.getURL('./siteBlockedAlert.html');` (the addon intern alert page).
3. Firefox do not visited the asked domain, firefox visited the rewritet addon intern domain.
4. So the displayed url is something like that: `moz-extension://9b334143-726d-4afb-8aa4-930b025ae4b5/siteBlockedAlert.html`

This is the code for the rewrite:
```
/*
 * Cancel the Request and redirect to the site blocked alert page,
* to inform the user about the full url blocking.
*/
if(result.cancel){
     return {
          redirectUrl: siteBlockedAlert
      };
}
```
And will be evaluate here **on line 7**:
```
/**
* Call by each Request and checking the url.
*
* @type {Array}
*/
browser.webRequest.onBeforeRequest.addListener(
    clearUrl,
    {urls: ["<all_urls>"]},
    ["blocking"]
);
```

**Comment by Thorin-Oakenpants** [comment]
Thanks for the info :+1:  I saw an image on AMO (the third one) and **to me** it looked like the alert was placed **in** the web page in lieu of a blocked element or something - it doesn't reach the edges of the page. Sorry if the term "inject" is not correct (told ya I was a dummy) :)

**Comment by Thorin-Oakenpants** [comment]
Added [Forget Me Not](https://addons.mozilla.org/firefox/addon/forget_me_not/) | [GitHub](https://github.com/Lusito/forget-me-not) to the list above to keep an eye on etc.

This **may** get closer to the all-in-one persistent storage extension I am dreaming of. Blacklist all (delete after domain no longer open), whitelist (keep), greylist (session only) etc. But it has issues due to FF limitations such as FPI, containers, PB mode, localStorage by host (FF58+) being the main ones.

It does have some extra items such as history, form data, service workers, certs, IDB - but they're listed under clear all. So far the ONLY thing cleaned by domain is cookies (and localStorage in FF58+), and you can do that with C-AD. Still, worth keeping an eye on maybe.

**Comment by Atavic** [comment]
[Nano Adblocker](https://github.com/NanoAdblocker/NanoCore) *Beta* is out, I hope someone will test it as I do.

**Comment by mazesy** [comment]
Concerning the [WebApi Manager](https://addons.mozilla.org/en-US/firefox/addon/webapi-manager/) Firefox extension in the context of sites using IndexedDB when cookies are not blocked (Issue #294), among all its features there is one which deals with that, listed in the extension's Options / Storage / Indexed Database API item: blocking it forbids sites from using IndexedDB without altering Webextensions which need it, which is interesting at least until the arrival of Firefox 59 which is announced as resolving this issue.

**Comment by verlain3** [comment]
Does Cookie AutoDelete works now with LocalStorage on Firefox 58 that came out today?

**Comment by mazesy** [comment]
> Does Cookie AutoDelete works now with LocalStorage on Firefox 58 that came out today?
And the answer is... yes!

**Comment by Forsaked** [comment]
Does Request Control still work for you guys?
It looks like it stopped working for me weeks ago, even with a vanilla profile no user.js and no other extensions it doesn't do anything with it basic rules or custom rules.

**Comment by Atavic** [comment]
Request Control 1.7.1 on FF ESR 52.0 32-bit works:
https://github.com/tumpio/requestcontrol/wiki/Testing-links

**Comment by Thorin-Oakenpants** [comment]
1.8.6 on FF57 seems to work for me: my little imdb stripper rule worked. I do know it was broken for one release about 2 months ago

**Comment by crssi** [comment]
@Thorin-Oakenpants would you mind to share this little imdb stripper rule?

**Comment by Forsaked** [comment]
Cross checked with work Computer, works fine here but not at home.
Even with a new vanilla profile it doesn't seems to work there.
I hope it is not Malwarebytes again, which f***ed up my last profile 4 weeks ago.

**Comment by forteller** [comment]
I would like to recommend adding the extension [Redirect AMP to HTML](https://addons.mozilla.org/en-US/firefox/addon/amp2html/). Google is trying to gather as much information about everyones browsing habits as possible, and to eat up as many web services as possible. Now they've gone so far as to host other peoples content for them. This is very destructive to the web, and your privacy. Read more about it here: https://danielmiessler.com/blog/google-amp-not-good-thing/

This extension redirects you away from AMP, so you don't use it yourself and don't link others to it when sharing the link. It's very important to send the message to newspapers and everyone that we don't want Google to own the web, we don't want AMP.

Thanks!

**Comment by grauenwolfe** [comment]
[CSS Exfil Protection](https://addons.mozilla.org/en-US/firefox/addon/css-exfil-protection/)

Based on reading about this on Bleeping Computer earlier today. Anyone already familiar with this BS?

> With the recent upgrades to the CSS language, CSS code has become a powerful tool that could be abused to track users on websites, extract and steal data from a web page, collect data entered inside form fields (including passwords), and even deanonymize Dark Web users in some scenarios.
> 
> Over the past month, three interesting research projects have tackled CSS as an attack vector, showing how this once benign language could be turned against users.

Article on Bleeping:
[CSS Code Can Be Abused to Collect Sensitive User Data](https://www.bleepingcomputer.com/news/security/css-code-can-be-abused-to-collect-sensitive-user-data/)

**Comment by Thorin-Oakenpants** [comment]
^^ continue it here: https://github.com/ghacksuserjs/ghacks-user.js/issues/343#issuecomment-363645884

**Comment by Thorin-Oakenpants** [comment]
css exfil protection added to wiki (extensions & test pages) - my FF without the extension enabled was vulnerable only in same-origin, cross-origin seemed to be already covered - no idea what it is in a nilla FF

**Comment by crssi** [comment]
FYI:
If you block access to storage\temporary folder then you can't manage rules in "Header Editor" addon.

Cheers

**Comment by Thorin-Oakenpants** [comment]
Can you elaborate - what is blocking temp storage - is css exfil doing that?

**Comment by crssi** [comment]
No, no, you got me wrong. CSS Exfil doesn't have nothing with that.
Its general. The sub folder "storage\temporary" in profile folder.
I had for some reason (testing) for a long time now, deleted this folder and created a file with the same name, to block its usage. And in this case I can't manage rules in "Header Editor" addon.
The same I had done long time ago with "storage\default", but now, for obvious reasons how WEs storage works (, this is not a case anymore.

**Comment by Thorin-Oakenpants** [comment]
that's an IDB folder (but IDB can also be used by web workers - eg think of uBO's web worker when searching for the block via logger in the 3rd party filters). I can't find the doc (I swear it was in MDN docs) about temporary folder use. It was an entry about permanent/chrome/temporary - and from what I remember, the temp was nothing to worry about - I think it HAS to clean up after itself when finished. I should find that doc again.

**Comment by GitCurious** [comment]
For those who manage cookies:  **CookieBro**
https://addons.mozilla.org/en-US/firefox/addon/cookiebro/

I am only 'testing' Quantum in portable mode [mainly testing CSS replacements for CTR etc]
as my daily browser is Waterfox 56 - yet I have also replaced my beloved [but now legacy] CookieKeeper with this addon in Waterfox as well.

Since yesterday (v2.0.1) _individual_ cookies can now be whitelisted and protected from deletion - this was my one proviso before changing cookie managers.

It has many other features but critically (for me) it is very easy to use in addition to having individual cookie protection.

I save a few logins & prefs and auto-delete the rest.

**Comment by mazesy** [comment]
For those who manage cookies: CookieBro

I discovered CookieBro yesterday, had a look, not bad, but incompatible with FPI First Party Isolation, and this is of course specific to FPI at this stage of Firefox (I think FF59 will make it possible, not sure).

Unfortunately CookieBro doesn't handle setting a site's cookie authorization to Session only.
Take the scenario of a user who wishes to have all cookies blocked with exceptions for a few sites. The developer of CookieBro says to include *.* in the extension's Blacklist option and then to set a whitelist exception for a given site : but what happens if my wish is not to have a site whitelisted but only accepted for the session? i.e. userstyles.org : the site requires a cookie just to display, otherwise white page! I'm not going to whitelist userstyles.org just to be able to access it -> a sesssion-only cookie is the solution.

Be it reminded that with CookieBro, whitelist = allow, blacklist=block ... but session exception is missing.
That doesn't do the job completely.

**Comment by GitCurious** [comment]
Cookiebro 2.2.0 - 2018-03-17
- Feature: you can now allow blacklisted sites to set session cookies. 
Off by default. (enable in Options)

**Comment by Decopi** [comment]
I would like to suggest the "Temporary Containers" add-on by stoically:

[https://addons.mozilla.org/en-US/firefox/addon/temporary-containers/](url)

I was a container-user without add-ons. I generally believe that the less add-ons, the better.
However, Container in Firefox is too basic/primitive in the sense that it has layout options but not functionality options. Even the Firefox' container add-on is very limited in functions. Everything related to containers in Firefox has to be manual, each website is handled manually (a nightmare).
I hope to be wrong, but with this lack of functions, average users are going to use Firefox' containers in the wrong way.

For example, containers in Firefox have not the option to open every website in new separate different container. Neither have the option to clean the container when closing (after "x" minutes).
In terms of privacy, this two functionalities are a must! Otherwise, is non sense to use only one container, the same container, for all websites. And the second function deleting everything, it avoids the need for other add-ons like cookie add-ons etc.

The "Temporary Containers" add-on by stoically has this two functions and many other functions.
Also, is an Open Source:

[https://github.com/stoically/temporary-containers/issues](url)

And "Temporary Containers" is just 369.5 KiB (only 100 KiB more than Firefox add-on).

**Comment by Thorin-Oakenpants** [comment]
> Neither have the option to clean the container when closing (after "x" minutes)

I need to investigate this. PB mode (which does not allow IDB) can destroy all PB mode related data (well, it's supposed to - cookies, localStorage, appCache, cache, service workers cache, asm.js cache etc).
 
Can a container window (which does allow IDB) actually clear only it's IDB stuff? I do not think so!!! How a container clears selective data depends on how it's doing it. How does PB mode know what to clear. I am thinking that extensions such as Temporary Containers are using the same APIs etc that other cleaners are using - eg Cookie Auto-Delete. And many persistent storage items do not have sufficient APIs to clear by host or time-range.

**Comment by mazesy** [comment]
Containers : I've never used them, I even block all privacy.userContext.xxx
Private Browsing (PB) : I never use it
FPI : yes, yes and yes. Very strong, very.

Nice resume, @Thorin-Oakenpants , clear enough (especially for a Sunday morning!)

**Comment by Thorin-Oakenpants** [comment]
^^ TL;DR

This is all you need IMO
- Use FPI
- Use a one off PB window for logging into different accounts on the same service (added bonus, no IDB), as this guarantees no persistent data

But...
- Use containers (FF shipped setup) if you want, on top of FPI, if can see any extra benefit in it (I can't)
- For sites that cross-domain logins are an issue, use a portable Chrome or a secondary Firefox profile. I am not sure of how widespread this issue is. Or consider not logging in or dumping the service. If you take privacy seriously, then this sort of sh*t shouldn't be encouraged

And...
- extensions that **claim** to clear (all) data can be misleading, and give a false sense of privacy by leaving behind (orphaned) tracking data
- That said, FF are working hard on Storage Management v2, Quota Manager, a Site Data UI, improved sanitizing, better integration with OAs, and even better wording in the UI and a dozen other aspects
- Hopefully one day, EVERY single type of persistent data can be cleared by a single API call by host or time-range and bulk cleared by OA = the holy grail IMO.

**Comment by Thorin-Oakenpants** [comment]
> clear enough (especially for a Sunday morning!)

f*k .. I ended up writing a book

> Nice resume

The word is summary.. a résumé is personal and outlines said person's background and skills. Maybe I should have done that instead, it'd be shorter :grin:

**Comment by Decopi** [comment]
Hi @Thorin-Oakenpants ! Thanks for your comments.
I am sure that the author of "Temporary-Container" (@stoically) is the right person to address your comments. His add-on' page also includes illustrative texts. But please, allow me to share my opinion:

FPI is great, and the intention here never was to use Containers instead FPI!!!!!!!!!!!
Both (FPI/Container) are two different things, created for two different needs.
In fact, both are complementary, and in my opinion both should be used together for better (not best) privacy practices/benefits.

I agree with you about "containers not mature". But that is the reason I recommended @stoically add-on. At present time is the only add-on that today expands container functions. I am not saying that this is the holy grail. I am just suggesting this add-on because increase privacy/security, and because is the best option out there for containers.
I understand that you don't see a real need for container. I was in contact with Mozilla' employees and other experts in FPI/Containers, and I am convinced about the utility. But as I said, the author of the add-on (@stoically) is the right person to address your arguments.

I also understand your general idea to replace containers by using PB tabs in normal mode. The problem is that for average users is unpractical. I know that your approach regarding privacy/security always is the best and the right to do. But sometimes we need to think in average users, and we need to think about alternatives (not solutions) for average users. Even if these alternatives are not solutions, are not perfect, are best and not better practices, if these alternatives increase privacy/security... will should use it.

**Comment by mazesy** [comment]
My problems with English. I did mean _nice summary_ ...

It's not a book. Modern trend to squeeze and compress has its limit; when the topic is complex compressing exaggeratedly would inevitably omit the necessary info.

The topic is complex, the summary clarifies the ingredients, for me anyway.

**Comment by Thorin-Oakenpants** [comment]
Lets just clarify a couple of statements here

> FPI is great, and the intention here never was to use Containers instead FPI

But if you're using FPI, then Containers often NOTHING you haven't already got, just stating the facts

> In fact, both are complementary

There's nothing complimentary about them - FPI effectively **overrides** containers in the sense that the "container" is now useless as a data bucket, because everything is already isolated on a more granular level.

If you are talking about data cleaning, that's nothing other extensions can't already do (eg Cookie Auto-Delete), and already do in normal mode (I am not 100% certain that some ContextId data types that can't be cleared by host, can't be cleared by OA, so correct me if I'm wrong).

---

Yup, I know not everyone has the same strategy. For example, I block all cookies except for about 8 I keep of which 4 are for logins, and another 4 or 5 which are session only (to make some sites work). Those 4 logins, I do not care about 1st party tracking (i.e linking previous visits) because I logged in, they already know me by a unique id (an exception here would be multiple accounts on the same service, which I do not have = I would use the PB window trick).

Because cookies control 1st party localStorage and IDB (and because 3rd party cookies are blocked as well it blocks 3rd party access to almost all persistent data), because appCache has to ask for permission, because Service Workers are disabled .. I do not really have to worry about clearing data when I close a domain. That is, I do not have to really worry about `1st party` tracking. And I clear almost all data on close (eg session cookies, IDB etc)

Until sanitizing gets it's cr*p together, this has been my strategy and that of the user.js since inception - do not collect anything unless it is absolutely required. But that's ME.

I have no issues with Containers, or extensions that enhance that in some fashion. Because not everyone will want to enable FPI, not everyone blocks all cookies by default.

At some stage, maybe I'll add a recommendation for those who do not use FPI, to consider TC - but first I want (and do not have the time) to investigate what it says it clears etc. In the meantime its added to the first post

**Comment by Thorin-Oakenpants** [comment]
^^ don't get me wrong, I just had a quick look at the AMO page. Its a great extension (and the idea is great), but its also incomplete (FF lacking APIs). I just need to "vet" it first .. so for now, its on the list, not on the wiki

**Comment by Decopi** [comment]
Hi @Thorin-Oakenpants ,

I read carefully all your comments. Again, thank you!
I never will say you are wrong about FPI/Containers. But I will just say that your info doesn't match with the info I received from Mozilla's employee & other experts.

My understanding still is that FPI and Containers are two different things, with two different functions, for different needs etc... one is not better than the other... one is not replacing the other... and both are complementary. More important, my understanding is that the combination of FPI and Containers increases privacy/security, not in a perfect way, but in a very useful way for average users (not for expert users like you).

However, my original intention here was just to suggest an add-on. Period.
I am not here to debate the concept of FPI or Containers. This is a very different conversation that should be kept only between you and expert tech guys.

Thank you again!

**Comment by Thorin-Oakenpants** [comment]
> But I will just say that your info doesn't match with the info I received from Mozilla's employee & other experts

In what way? Without wanting to misquote anyone, what I have said on OAs is based, among other things, on chatting with Arthur Edelstein (key lead on the Tor Uplift) and he assures me, that while he's not 100% knowledgeable on all aspects of the coding, he was assured, that what he told me was correct, directly from one of the guys responsible for it at Mozilla (i.e the coder) - horse's mouth and all that.

As to their purpose and limitations - well of course they're different. One is superior/more-hardened for TBB, the other is something that grew out of an old add-on idea. The fact they use OAs is about the only thing in common.

> and both are complementary. More important, my understanding is that the combination of FPI and Containers increases privacy/security

Please stop saying they're complimentary. In almost everything I have said, I am talking about containers. I think you might be saying containers but thinking about "Temporary Containers". Containers on their own do NOTHING more than FPI already does (and a PB mode window when needed). I also wouldn't include the word security - this is all about privacy.

I get where you coming from. I too am excited by new privacy measures. Having now taken a break and come back to it (new topic, yet to read), something crossed my mind. And this comes back to the `1st party tracking` I was talking about earlier (i.e when you revisit a site in the same session), I would have to check, but if Temporary Containers assigns a new ContextId, then that would mean each instance of a same-domain visit would be a new isolation (obviously no existing tab of that domain should be open). So yes, if leveraged that way, then there is one example I can think of that this exceeds what FPI does, and since FPI also isolates other things, then in this ONE case so far, they're complimentary.

Not knocking anyone or anything here. Appreciate everyone's time and input :kiss:

**Comment by Thorin-Oakenpants** [comment]
note to self: https://addons.mozilla.org/firefox/addon/https-by-default/ is now a web extension

**Comment by Decopi** [comment]
I would like to submit to be added to the list, @Cowlicks add-on (Privacy Possum): https://addons.mozilla.org/en-US/firefox/addon/privacy-possum

The GitHub page (https://github.com/cowlicks/privacypossum) offers a nice description, also comparing PP with other similar privacy add-ons/extensions, showing PP advantages, added values etc.

**Comment by Thorin-Oakenpants** [comment]
Just some quick notes:

1. cookie tracking - we already block 3rd party cookies. Also FPI isolates 3rd party cookies if you allow them, same with containers (per container). And uMatrix allows cookies blocking with whitelisting on a per scope level, and then there's Cookie AutoDelete and a hundred other cookie extensions

2. ETags - we can already do this with a [rule](https://github.com/ghacksuserjs/ghacks-user.js/wiki/4.2.4-Header-Editor) for [Header Editor](https://addons.mozilla.org/firefox/addon/header-editor/) - or users can disable both disk+memory cache. The first solution is the best. The second can be a tad extreme (nothing in memory will cause extra traffic etc). A third solution is clearing cache eg say every 15 minutes via an extension but is not a perfect solution (time factors leave holes for repeat visits).

3. Referers - Smart Referer is a dedicated referer extension that allows a per origin-destination level of control, with whitelists/greylists/blacklists etc (not actually using it myself, so slightly hazy on it)

4. "fingerprinting" - uM (and uBO) can handle 3rd parties if used in "medium" or hardened mode or whatever they call it. Yes, not everyone uses it that way. And some 3rd parties are unavoidable no doubt (for the site to work). And there are 1st party FP scripts. Been subscribed to FPjs2 for years. Mozilla's RFP pref defeats most of this (fonts, multiple window metrics, hardware, timing, etc etc). The key to FP'ng is that it is a worst case scenario (always do your tests allowing all JS etc), but in really you control most if it by limiting JS and 3rd parties.

Not entirely sure what this extension brings to the table, except duplication and possibly interference with other extensions/methodologies (eg is talks about faking FP'ng stuff). Note: I have not installed it or checked it out (eg options).

**Comment by grahamperrin** [comment]
Maybe add [Privacy Settings](https://addons.mozilla.org/addon/privacy-settings/) (@schomery) to <https://github.com/ghacksuserjs/ghacks-user.js/wiki/4.1-Extensions> …

**Comment by grahamperrin** [comment]
> … what the user.js already covers …

I assumed that the 'Extensions' page includes extensions that might be used in isolation (not necessarily with the `user.js`).

**Comment by Thorin-Oakenpants** [comment]
I get your point. Most extensions that do things FF can't eg in legacy FF there is no way to hide resource URIs, or in FF57+ how do you handle CSS Exfil, how do you do granular control of referers with a pref (you can't), or how do you build smart rule sets for HTTPS. etc

Since we're focusing on non-legacy extensions, Privacy Settings offers 6(?) pref tweaks (probably at least two more with FPI and RFP), and in the overall picture does almost nothing TBH - not knocking what Schomery has done at all, and AFAIK it's been abandoned (due to limitations in Web Ext). To me it's a false sense of privacy (but not due to the author)

The user.js on the other hand, offers [250+](https://github.com/ghacksuserjs/ghacks-user.js/issues/378#issuecomment-373927462) settings that differ from default, and enforce another 90+ at default values. While it might not have a pretty interface, I do not wish to recommend an extension that may not be maintained anymore, and can confuse users (eg if its in the user.js at true, but they keep setting it to false in the extension, and so on)

**Comment by grahamperrin** [comment]
Thanks for clarification. 

> … 26 items on the AMO page's first image, but the description only lists 16 "items" …

At a glance: the first image is probably of a legacy version, something less than 0.2.6.

**Comment by claustromaniac** [comment]
A handy one for security was updated to WE three days ago: [VTzilla](https://addons.mozilla.org/en-US/firefox/addon/vtzilla/). I only barely skimmed the source code, but the privacy policy is fine. 

Default settings kinda suck though, since everything you download is sent to VirusTotal. I would only enable the option to ask to scan the file on VirusTotal before downloading, and only use it to scan executables that I don't mind the VirusTotal community getting their hands on.

I realise this probably doesn't belong in the wiki, since this doesn't bring anything new to the table (nothing other security solutions can't cover), no real _hardening_ going on either, and even Firefox itself has its own built-in Safe Browsing for this purpose. I just wanted to let folks know.

**Comment by Thorin-Oakenpants** [comment]
^^ without having read it fully... so context menu: scans the file you point at on the server .. and THEN you can download it if you want after the VT result. Seems like a reasonable extension. But as you said, the default settings suck and I wouldn't want to promote it for that reason (because people are lazy and just blindly install things without reading). Maybe the dev(s) can be persuaded to change that

That said, I agree this probably outside our scope. Not my mandate to babysit people's DL and web habits

**Comment by claustromaniac** [comment]
Just to make clear why I still wanted to share it here (for anyone wondering):

I personally find VirusTotal, HybridAnalysis and similar services really useful because I normally prefer to tighten security where data comes in, instead of continuously scanning memory and file system for malicious code (the latter being what resident antiviruses do). Why? The same reason I wash my hands often instead of leaving everything to my immune system (I don't have OCD... not THAT often).

When you mix these services with healthy practices and tools such as a decent firewall, keeping the whole system up to date, being cautious with what you plug in all your ports (and tweaking the system to disable autorun and such), and running in a virtual machine the shit you still just can't trust, then you don't need resident antiviruses. They become mostly just a waste of system resources.

It is probably not an optimal mindset for the average computer user (some of said practices require a fair bit of know-how, albeit nothing is really difficult), but I suspect I'm not the only one with that philosophy around here. :)

**Comment by crssi** [comment]
@claustromaniac nice one...
`Show 'Send to VirusTotal' prompt when downloading files` and `Pause downloads when sending to VirusTotal` options doesn't seem to work (haven't tried much now, but maybe containers), otherwise unchecked all options, having just a short-cut from context menu is nice.

**Comment by Thorin-Oakenpants** [comment]
> but I suspect I'm not the only one with that philosophy around here

I think among users here you would be in the majority. Any decent strategy would cover all layers/levels/architecture and have multiple fallbacks. As long as you're informed, you can do what you like based on your threat model / risk assessment.

Like you, I prefer to be proactive. AV is massively overrated IMO (although it's mandate has changed a lot over time). To me, it's something that comes too late. I'd rather plug all the holes that could let it get on my system in the first place. Not for everyone, but I actually don't even bother having AV at all, I feel that secure in my setup. In the 20 years or so I had an AV, I never once had anything except false positives.

The biggest problem in security is ... humans. People are inherently stupid, and you can't (usually) fix stupid. This is why OpSec is so hard to master/teach

**Comment by bberberov** [comment]
I tried [Forget Me Not](https://github.com/Lusito/forget-me-not) and it is not removing cookies in the current version.  Waiting on Lusito/forget-me-not#59.

**Comment by crssi** [comment]
@bberberov 
Try [C-AD](https://addons.mozilla.org/firefox/addon/cookie-autodelete/).
Is with C-AD the same problem?
If it is, I might have one idea.

**Comment by Thorin-Oakenpants** [comment]
I'm not using any cleaning extensions (I just don't let anything get stored really) - might take a while for everything in that regard to mature due to FF web extension limitations etc - look at all the persistent storage types (cookies, localStorage, IDB, app cache, SW cache) and mix that with  OA's (FPI & PBMode & Containers) and between all the combos, there are still issues.

**Comment by Joel889** [comment]
https://github.com/Lusito/forget-me-not works if set up correctly. There is also a nice update coming soon. I tried C-AD, but stopped using it, because it doesn't clean Local Storage and Indexed DB well. The developer keeps saying it cannot be done, but somehow ForgetMeNot not is doing it.

**Comment by crssi** [comment]
@Thorin-Oakenpants  I know, you are a big girl who can take care of mostly all of the internet browsing shit and I do agree with you completely but only when we are talking about us... geeks.
I do this stuff also for elder users, who does not undertand 1% of this shit storm and for them is essential that 99% stuff works without any user interactions.

Quite a lot pages elder people are using does not work when cookies are denied, specially 1st party.
There are a few that doesn't work if 3rd party cookies are denied (for example vimeo embedded video), funny is that those are still working when you just accept 3rd party cookie and do a outgoing deny with uM,

So what I do:
FF accept all cookies... dealing with C-AD for first party and deny all 3rd party with uM.
What I have found out that C-AD does not work as intended when uM->Settings->Privacy the option `Delete blocked cookies` is ticked/enabled. That is what I assumed @bberberov might have the problem.

@Joel889 in the past "Forget Me Not" didn't clean LS as well (this might changed lately, but I haven't keeping an eye on it... will give today another look). What does exaclty mean "if set up correctly"?

For sure there is another WE... "Temporary Containers" which is really great, but for real it has to polish some quirks to be really set and forget and I hope author will find some time to do those.

Cheers

**Comment by claustromaniac** [comment]
> Show 'Send to VirusTotal' prompt when downloading files and Pause downloads when sending to VirusTotal options doesn't seem to work (haven't tried much now, but maybe containers)

True. I didn't test much either, but it could be just a bug. It _is_ tagged as experimental, after all.

> AV is massively overrated IMO

I mostly agree with you. If you're ahead enough in the learning curve you can perfectly ditch AV entirely, in that sense AV is overrated. Not to mention that hash-based detection is very easy to fool, and heuristics has many pitfalls.

I still think they're necessary (and good) tools because not everyone can afford the time (or brains) to learn what's needed to feel that safe without one. Besides, there _is_ a learning curve (no one is born knowing shit), so even people with the time and brains to learn are better off using one while they learn.

Anyway, I better drop this before going off topic much more lol. Sorry about that.

By the way, [I suggested the VTZilla devs to change the defaults](https://addons.mozilla.org/en-US/firefox/addon/vtzilla/reviews/). Let's see if that argument is good enough for them.

**Comment by bberberov** [comment]
@crssi I don't have `Delete blocked cookies` on.

@Joel889 I just went Quantum from being on ESR ahead of ESR changing at 60.  I used CookieKeeper etc.  I didn't use to keep cookies, but they I started using Bugzillas more, and it's very convenient to configure the view once and have it persist, even when not logged in.  I think that's a sensible use case that I shouldn't have to give up on.  There are others.

@Thorin-Oakenpants I'm not a dev, but I looked at some of the WE API, and there does not seem to be a lot to work with there.  I can see now why devs are not porting extensions and functionality is lost.

**Comment by Thorin-Oakenpants** [comment]
@bberberov I didn't mean it that it is worse now than it was before.

I have never used a cleaning extension, so I'm not 100% sure - and due to my lack of allowing 99.9% of persistent data in the past, I have no anecdotal evidence (eg until FF56 IDB was disabled, until 59 appcache was disabled, until 60 workers were disabled - cookies remain so - I've never seen an IDB entry in my life!) ... but I believe with IDB that there has NEVER been the ability for sanitizing it by host or time range. Also FPI (FF51) & cookies are an issue (fixed in 58). So currently ESR52 users with FPI have issues (i.e I think even legacy extensions can't handle OA's). If anything, the new Web Extensions API will allow all of this to be properly sorted out once and for all

**Comment by mrdokenny** [comment]
@Joel889 
> I initially used C-AD, but stopped using it, because it doesn't clean Local Storage and Indexed DB well. The developer keeps saying it cannot be done, but somehow ForgetMeNot is able to clean everything. If you read through my posts, I've tested this extensively with both addons.

CAD already clears localstorage if you enable the setting.

As for as I know, there's no API to clear [IndexedDB](https://bugzilla.mozilla.org/show_bug.cgi?id=1340511) **per individual site**. Right now, you can clear IndexedDB for **every site** as well as various other data from the browsingData API. I could implement it with all those other data but it would be copying every other simple cleaning extensions that do the exact same thing. [One example](https://addons.mozilla.org/en-US/firefox/addon/clear-browsing-data/)

I rather use APIs that can take advantage of individual site cleaning rather than an all or nothing approach (since you can just use various other simple cleaning extensions that are *easy to make* or use the **Clear history when Firefox closes** option).

**Comment by crssi** [comment]
@Joel889 
Forget Me Not does not clean IndexedDB (only manually for all sites as @mrdokenny say) and for cookies it work very unreliable, where CAD does the job very good.
CAD wins over FMN big time.

**Comment by bberberov** [comment]
@Thorin-Oakenpants This is slightly off topic, but in your opinion, can `Places` be re-implemented as a Web Extension?  Is there enough access through the API to record the same complete history and a storage capability to keep a bookmark hierarchy?  I looked a little at the documentation, but I'm new at this.

**Comment by Thorin-Oakenpants** [comment]
@bberberov I am not an extension developer, so I have no idea. There should be storage capacity for web extensions (eg IDB) but size limits may play a factor. I'm not sure why you would want to reinvent the wheel - eg FF already collects all this info - is it for display purposes? Maybe an extension can hook into places.sqlite and display/search it in a shiny new interface. I seem to remember a legacy extension or trial (but may be confused with the browser involved) that showed (based on history) things like frequency of visits, number of visits, etc with little graphs and stuff on a dashboard.

**Comment by bberberov** [comment]
@Thorin-Oakenpants Annotations are going away, which includes descriptions:
https://bugzilla.mozilla.org/show_bug.cgi?id=1460577
The WE API for bookmarks is limited.  There are new regressions in bookmark handling in 59/60 vs. 52.  I'm looking for a solution to these issues.

**Comment by forteller** [comment]
No one has said anything about my comment above. IMO it's a very important privacy extension, so let me quote myself:

> I would like to recommend adding the extension Redirect AMP to HTML. Google is trying to gather as much information about everyones browsing habits as possible, and to eat up as many web services as possible. Now they've gone so far as to host other peoples content for them. This is very destructive to the web, and your privacy. Read more about it here: https://danielmiessler.com/blog/google-amp-not-good-thing/
> 
> This extension redirects you away from AMP, so you don't use it yourself and don't link others to it when sharing the link. It's very important to send the message to newspapers and everyone that we don't want Google to own the web, we don't want AMP.
> 
> Thanks!

**Comment by Thorin-Oakenpants** [comment]
^^ not forgotten .. here's my inbox (somewhat sanitized!)
![notforgotten](https://user-images.githubusercontent.com/16656956/40139920-eac0d83e-5940-11e8-96f5-803b94f47f89.png)

The top line just came in. I still have, marked as unread (but I read it) the one from 3 months ago yet to action. That is, it's still actionable because I haven't written it off, so to speak. I agree it's important to stop this BS, but at the same time, this repo is predominantly about desktop - as in I do not wish to end up dealing with Android BS if I can help it.

That said, what does everyone else think? It's definitely a privacy issue, but does it really give any added privacy on Android? Note: I haven't looked at the actual extension. Maybe start a new issue, IDK.

**Comment by earthlng** [comment]
@forteller thanks for mentioning it again, I totally missed your original comment.

AMP sounds absolutely appalling but what else can you expect from Google. I completely agree with everything written in your linked article. Thanks for the link

> This extension redirects you away from AMP

unfortunately only after you visit the AMP site because it has to get the non-AMP link from the source code. Still useful when you bookmark a site though, for example.

> and don't link others to it when sharing the link.

this is probably the main reason to use the extension. It's not just about mobile either, Pants.

---

github: https://github.com/da2x/amp2html

it's a single contentscript injected into all http(s) sites and runs at document-end.
There are 3 parts to it:
- redirect away from AMP sites
- modify links on news.google.*
- modify links on google search

I never use anything google and don't share a lot of links so I don't have much use for the extension but if it can help that potentially fewer people link me or anyone else to AMP and thus google servers, I'm all for recommending it. Anything we can do to fight back against the evil G gets a :+1: from me

**Comment by Joel889** [comment]
@mrdokenny 

> CAD already clears localstorage if you enable the setting.

Not always as confirmed here:
https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/issues/355

Both localstorage values (https://html5demos.com/storage/) disappear automatically in ForgetMeNot, but one of the values remain in C-AD.

Yes, IndexDB cannot be cleaned on a per site basis, but having an on demand button beats having to close the browser in order to clear it.

**Comment by 2glops** [comment]
Good news for SSleuth (the only legacy extension I missed) :
github.com/sibiantony/ssleuth/issues/78

**Comment by claustromaniac** [comment]
Quoting myself:

> I suggested the VTZilla devs to change the defaults. Let's see if that argument is good enough for them.

Welp, it clearly wasn't. They changed its privacy policy instead :man_facepalming:. More precisely, they outright removed the one that was on AMO and lumped the bits relevant to the extension with the general VT privacy policy. Not only that, but they also added a new opt-out exclusively for collecting information.

Moreover, quoting the relevant part of the policy:

> Existing users of a VirusTotal extension will need to opt-in to share pDNS data with the Community. Users downloading the VT extension for the first time may opt-out of this collection in the  extension’s settings.

...the funny thing is that's not even true. Existing users have to opt-out too. This move has EvilCorp written all over it. I very much doubt something like this would've happened back when VT was not owned by EvilCorp.

I guess I'lll go back to bookmarking VT and manually scanning files when I want to. I don't want to trust them and I don't wan't to dissect the extension on every update just to see if I keep it or not. Not worth my time. 

Feel free to move this under Rejected in the OP, if you want.

**Comment by crssi** [comment]
^^ But you can still uncheck all and have a nice shortcut with context menu to virus scan on demand.

**Comment by claustromaniac** [comment]
@crssi sure, I can do that, but did you see what the policy looks like now? They don't make very clear what parts of the policy apply to the use of the extension and what parts don't. The policy on AMO used to be very specific on that. As it looks now, they could push an update any day adding data collection under the hood and I won't be sure they aren't doing that if I don't look at the source code first. 

I just don't want to trust them, and I don't want to invest that kind of time just for a context menu entry. Even writing my own extension for that would be a better option than revisiting the source code every now and then lol.

**Comment by Atavic** [comment]
Virus Total's good for Google to collect info, while the protection/warning issued don't cover 100% the VT user. Malware creators use different services that are similar to VT but do not share (at least not openly) the results of file scanning.

I'd use Kaspersky instead, while keeping in mind that you're covered on 90% of online malware.

**Comment by laserstrike** [comment]
Blend In?
https://addons.mozilla.org/en-US/firefox/addon/blend-in/

This is the only user agent spoofer I found that *passes* platform and oscpu at https://browserleaks.com/javascript. 
I don't know if the fact that there's only one setting is desirable, though.

**Comment by Thorin-Oakenpants** [comment]
^^ Thats OK. It's interesting from a technical perspective. Just been reading about it - https://github.com/haqer1 is the owner so we could always @ him/her. I was surprised to see of the few reviews that they go back 7 years! 1000 users. Never heard of this before. I'm semi-impressed by the buildIDs to be honest. Interested to know how/who/what maintains the various spoofs, but its a nice touch that its plausible.

A decent UA spoofer could be handy, if used in moderation. AFAIK, an extension can override RFP's spoofing (since its the last one to modify the header?, NFI about JS stuff) - so I can see edge cases where you might spoof as a different version. It kinda needs a global pref to spoof-or-not and then a white or black list. And it kinda needs to allow end users to define the UA.

Anyway, end of the day, you cannot truly hide your OS, but you can probably fool most sites/FP tests - since they don't (yet) get really down and dirty. So IDK, a OS+version spoofer seems fairly unimportant except to *unbreak* sites, not as a mechanism to defeat FP'ng.

**Comment by Thorin-Oakenpants** [comment]
^^ Since the RFP spoof doesn't spoof the OS (except to 4 platforms), this means Linux tends to stand out. And since "advanced" users (think uMatrix, uBO etc in hardened mode) can really limit exposure, this could a handy tool (i.e a decent UA spoofer) to handle the case of OS/platform spoof

https://hardware.metrics.mozilla.com/ - Win 7 43%, Win 10 38% ... Linux 2.7%

**Comment by Atavic** [comment]
[containerise](https://github.com/kintesh/containerise) on [AMO](https://addons.mozilla.org/en-US/firefox/addon/containerise/) automatically open websites in a container.

**Comment by Atavic** [comment]
[Containerise](https://github.com/kintesh/containerise) on [AMO](https://addons.mozilla.org/en-US/firefox/addon/containerise/) automatically opens websites in a container.

**Comment by curiosity-seeker** [comment]
> Blend In?
https://addons.mozilla.org/en-US/firefox/addon/blend-in/

> This is the only user agent spoofer I found that passes platform and oscpu at https://browserleaks.com/javascript.

Yes, but it's still not perfect: on http://ip-check.info/?lang=en the _User-Agent_ is reported as Windows NT 6.1 and the _Browser Type_ as Mozilla 5.0 (Windows). However, the _System_ is (correctly) reported as Linux. So this confirms Pants' remarks: you cannot reliably hide your OS.

**Comment by crssi** [comment]
@Atavic it looks Containerise is better alternative to Multi-Account Containers, at least permanent containers created within is not deleted by TC... at least from what I see.

**Comment by curiosity-seeker** [comment]
@crssi 
> it looks Containerise is better alternative to Multi-Account Containers, for example permanent containers created within is not deleted by TC

I've never seen this on my system. Permanent containers in MAC have never collided with TC so far.

**Comment by crssi** [comment]
If permanent container is created in FF settings then not, but if it created over MAC, then after TC cleansing those are gone... see `https://github.com/stoically/temporary-containers/issues/157`
When using Containerise instead of MAC, those problems are gone.

**Comment by curiosity-seeker** [comment]
> If permanent container is created in FF settings then not, but if they are created over MAC, then after TC cleansing those are gone

Well, not here. I always create them in MAC.

**Comment by crssi** [comment]
Which FF version do you use? I am on 60.0.1.

**Comment by curiosity-seeker** [comment]
So am I.

**Comment by crssi** [comment]
Can you reproduce it on a clean profile?

**Comment by curiosity-seeker** [comment]
No, I haven't tried that yet. I'm afraid we're getting a bit OT here ;-)

**Comment by crssi** [comment]
Sure we are... sorry, I will delete my clutter now.
Cheers

**Comment by KevinRoebert** [comment]
Hello. I just wanted to say that ClearURLs moved to GitLab. The repository can now be accessed via the following link: https://gitlab.com/KevinRoebert/ClearUrls

**Comment by Thorin-Oakenpants** [comment]
@KevinRoebert  Thanks. I've fixed our link. PS: in your readme notice your misspelled "Official" (missing the second `i` )

**Comment by Thorin-Oakenpants** [comment]
https://addons.mozilla.org/firefox/addon/universal-bypass/ - if anyone wants to check it out (excuse my ignorance on link shorteners) does this bypass giving away your browsing to the relevant parties (like the EvilCorp AMP bullshit)?

**Comment by claustromaniac** [comment]
> does this bypass giving away your browsing to the relevant parties

In short: yes. 

AFAIK It's merely a convenience thing. It helps speed things up a bit, and it _might_ help avoid unwanted side effects of clicking scripted buttons on those sites (like redirections and whatnot). The extension still needs you to allow scripts on most (if not all) of those URL shortening sites, though, so that last part might not be of much relevance anyway.

> (like the EvilCorp AMP bullshit)

If you're refering to the Convert AMP to HTML extension, I'd say that extension is just not meant to protect one's privacy in any way. It's a useful extension, just not privacy-wise.

**Comment by claustromaniac** [comment]
> I never click on shortened links anyway

Going a little OT, you might want to add [those domains](https://universal-bypass.org/example-links) to your uBO filters, just in case. Before checking out that extension I wasn't even aware there were _that many_ link-shortening sites out there...

**Comment by Thorin-Oakenpants** [comment]
> >  (like the EvilCorp AMP bullshit)

> If you're refering to the Redirect AMP to HTML extension

Nope. By bullshit, I mean 3rd parties sitting between me and the content... yet another asshole clipping the ticket

**Comment by claustromaniac** [comment]
I was going to make my own extension for this but I found one that looks good: [Detect Cloudflare](https://addons.mozilla.org/en-US/firefox/addon/detect-cloudflare/) | [GitHub](https://github.com/traktofon/cf-detect)

I was a little bit surprised when I found the Block Cloudflare MITM extension is not listed here. Should I have posted this in #310 instead?

I also found [Claire](https://addons.mozilla.org/en-US/firefox/addon/cloudflare-claire/), but I like the visibility of Detect Cloudflare and the fact that it lists all third party domains that go through Cloudflare as well.

**Comment by Thorin-Oakenpants** [comment]
detect cloudflare .. so 50% of sites I visit and growing - I think it's a losing battle TBH. Block Cloudflare is too extreme IMO - i think i've covered this before, but you can just block cloudflare in uM (right?, or is it not apparent and gets loaded via a redirect or something? sorry for using the correct technical terms for shit). 

Claire looks kinda cool - shows HTTP2 and IPv6. Maybe we should discuss all this in a new topic

**Comment by Thorin-Oakenpants** [comment]
since we block HTTP2 and will do so for IPv6 (and NFI what a ray thingy is) .. I kinda like Detect Cloudflare especially the distinction tween orange and red

**Comment by claustromaniac** [comment]
> I think it's a losing battle TBH. Block Cloudflare is too extreme IMO

I agree. Blocking is not for everyone. Not for many, even.

> you can just block cloudflare in uM (right?, or is it not apparent and gets loaded via a redirect or something?

CF in the broadest sense is a CDN. It offers part of its services by delivering content as a third-party (Decentraleyes covers some or all of these), and that part can be blocked with uM, uBO, etc. The worrying part from a security/privacy perspective is their service as a [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy). That's what people refer to when they say CF is MITM'ing users. CF stands _between_ the server and you, and some or all HTTPS traffic gets decrypted and re-encrypted along the way without your consent (but with the server's consent, since they are the ones using CF).

CF uses a number of custom headers, including the CF-Ray header, that can be used to detect its presence as a reverse proxy. That's what these extensions do.

I generally don't give a shit about CF because I don't use most websites I visit for personal stuff, but sometimes I want to know if CF is intercepting the traffic or not. Sure, I could just check the headers myself, but that gets tedious pretty fast. The extension not only saves me that trouble, but also works as a constant reminder that I shouldn't blindly trust the padlock next to my URL.

**Comment by Thorin-Oakenpants** [comment]
Am thinking we should add Bitwarden to the wiki - personally, I plan on wiping all my FF saved passwords and disabling it, and using Bitwarden instead. Haven't really looked at the whole thing, but FF's encryption of passwords isn't meant to be the greatest (but I do not consider my PC to be under any threat). No other password manager extension comes close to BW, right? Anyone got any opinions?

might need to retire this issue and start a clean new one, and review what's been covered in the hundred comments

**Comment by curiosity-seeker** [comment]
> No other password manager extension comes close to BW, right? 

Hm. There have been several concerns ([example)](https://palant.de/2018/07/09/is-your-lastpass-data-really-safe-in-the-encrypted-online-vault) about the security of LastPass. What makes you sure that Bitworden does it better?

**Comment by Thorin-Oakenpants** [comment]
My understanding is that the data is encrypted locally and no-one but you can decrypt it, so the cloud versions for syncing cannot be compromised. Maybe I need to actually read up on the exact features. Plus it's open source AFAIK

**Comment by curiosity-seeker** [comment]
Yes, this is exactly how Lastpass does it, too - see, e.g., https://www.lastpass.com/how-lastpass-works . And yet there seem to exist - or, at least, existed - implementation errors. They attracted attention because Lastpass is the most popular password manager. Who knows if Bitwarden is affected by similar flaws. 

I am not sure if you should recommend any cloud-based password manager. There are many users who refrain from using them because of security concerns.

**Comment by Thorin-Oakenpants** [comment]
Hmmm .. I thought LastPass had all the keys, so to speak. Something I did for a client a few weeks ago really led me to believe they could see it all (password info rendered into an https page - I think it was either the print output or an online vault page or something). I guess not. BTW, a quick search unearthed this

https://hackernoon.com/psa-lastpass-does-not-encrypt-everything-in-your-vault-8722d69b2032

Not that BW couldn't have issues as well. BTW, BW allows you to use any cloud, including your own (but yeah, most people couldn't do that and would be even less able to make it secure)

**Comment by fmarier** [comment]
> Lately, I've been trying out MasterPassword (GitHub). I like the concept a lot, but I wish it were possible to create longer passwords. It can at most generate 20-char-long passwords AFAICT. I wouldn't use it for anything super important, for that reason.

That concept is usually called "password generators". The adblockplus guy [reviewed the security details of all of the ones he could find](https://palant.de/2016/04/20/security-considerations-for-password-generators) and ended up [writing his own](https://addons.mozilla.org/firefox/addon/easy-passwords/).

**Comment by claustromaniac** [comment]
Thanks for the info, @fmarier !

**Comment by claustromaniac** [comment]
Just FYI, regarding the Detect Cloudflare extension that I mentioned previously, I released a fork of it with some minor changes [here](https://github.com/claustromaniac/detect-cloudflare-PA).

**Comment by Thorin-Oakenpants** [comment]
I prefer toolbar icons, so everything is visually in one place

**Comment by claustromaniac** [comment]
I prefer a Page Action for this one because I don't need to see the extension's icon when CF is not detected. It just wastes valuable space on smaller screens. 

I opened a pull request to commit the other significant difference to the original extension. If it is merged, the behavior of the icon will be the only difference left between the original and my fork. I considered making more changes but maybe some other time :)
